app/sweepers/kudos_sweeper.rb

```diff
@@ -11,7 +11,7 @@ def after_create(kudo)
 
     # expire the cache for the kudos section in the view
     # and for the full-page, expanded list
-    expire_fragment "#{kudo.commentable.cache_key}/kudos-v1"
-    expire_fragment "#{kudo.commentable.cache_key}/kudos/full"
+    ActionController::Base.new.expire_fragment("#{kudo.commentable.cache_key}/kudos-v1")
```

Comment:
It won't fix that unfortunately. This is for a different issue.

If we use plain `expire_fragment` for this PR, the errors we get are:
```
  1) ChaptersController show assigns @kudos to non-anonymous kudos
     Failure/Error: expire_fragment "#{kudo.commentable.cache_key}/kudos-v1"
     
     NoMethodError:
       undefined method `expire_fragment' for #<KudosSweeper:0x000055cb65a32718 @controller=nil>
     # ./app/sweepers/kudos_sweeper.rb:14:in `after_create'
     # ./spec/controllers/chapters_controller_spec.rb:199:in `block (3 levels) in <top (required)>'

  2) ChaptersController show assigns instance variables correctly
     Failure/Error: expire_fragment "#{kudo.commentable.cache_key}/kudos-v1"
     
     NoMethodError:
       undefined method `expire_fragment' for #<KudosSweeper:0x000055cb65a32718 @controller=nil>
     # ./app/sweepers/kudos_sweeper.rb:14:in `after_create'
     # ./spec/controllers/chapters_controller_spec.rb:209:in `block (3 levels) in <top (required)>'
```

Totally out of the left field. I think what happened was:

- There's a bug when using sweepers with Rails 4 and RSpec controller tests: `https://github.com/rails/rails-observers/issues/31` (no cross-repo linking please, GitHub).
- A workaround is to use `ActionController::Base.new.expire_fragment` instead of just `expire_fragment`. We fixed this for the other sweepers in the repo (#2917), so why not this one?
- We never have any RSpec coverage for the kudos controller until now, which means finally this line gets run in specs, revealing the error: https://github.com/otwcode/otwarchive/blob/e59b8eba05cb6630b6a01f956be74076d64291fa/app/controllers/kudos_controller.rb#L3

In conclusion, :joy:.

Reply:
You say 😂, I say, "I wish :sob_blood: were a default emoji."